lib/tokenize.cpp

```diff
@@ -2003,9 +2003,19 @@ bool Tokenizer::simplifyUsing()
         // Move struct defined in using out of using.
         // using T = struct t { }; => struct t { }; using T = struct t;
         // fixme: this doesn't handle attributes
-        if (Token::Match(start, "struct|union|enum %name%| {")) {
-            if (start->strAt(1) != "{") {
-                Token *structEnd = start->linkAt(2);
+        if (Token::Match(start, "class|struct|union|enum %name%| {|:")) {
+            if (Token::Match(start->next(), "%name%")) {
+                Token *structEnd;
+                if (start->strAt(2) == "{")
```

Comment:
I do not understand why we have a if/else here. it seems we could always locate `structEnd` with a loop.

Reply:
In the first case we know where the end is because we know where `{` is. This is the common case and is also the fast path.
In the second case we know where `:` is but we don't know where `{` is so the loop is there to find it. The loop is slower.